src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -151,6 +152,23 @@ public IIntellisenseResult Suggest(string expression, RecordType parameterType,
             return suggestions;
         }
 
+        /// <summary>
+        /// Creates a renamer instance for updating a field reference from <paramref name="parameters"/> in expressions.
+        /// </summary>
+        /// <param name="parameters">Type of parameters for formula. The fields in the parameter record can 
+        /// be acecssed as top-level identifiers in the formula. Must be the names from before any rename operation is applied.</param>
+        /// <param name="pathToRename">Path to the field to rename.</param>
+        /// <param name="updatedName">New name. Replaces the last segment of <paramref name="pathToRename"/>.</param>
+        /// <returns></returns>
+        public RenameDriver CreateFieldRenamer(RecordType parameters, DPath pathToRename, DName updatedName)
+        {
+            Contracts.CheckValue(parameters, nameof(parameters));
+            Contracts.CheckValid(pathToRename, nameof(pathToRename));
+            Contracts.CheckValid(updatedName, nameof(updatedName));
+
+            return new RenameDriver(parameters, pathToRename, updatedName, CreateResolver(Config.WithoutDisplayNames()));
```

Comment:
Why is .WithoutDisplayNames() being used here?

Reply:
added comment explaining the logic. We basically just need an environment where the only logical => display pair is the one that we're actively renaming.